File: research/benchmark/framework/problems/calculator/typescript/template.ts
export type Op = "add" | "sub" | "mul" | "div";

export type Expr =
  | { tag: "lit"; value: number }
  | { tag: "binop"; op: Op; left: Expr; right: Expr }
  | { tag: "neg"; inner: Expr };

export function evalExpr(e: Expr): number {
  // TODO: implement
  // Throw new Error("division by zero") for division by zero
  throw new Error("not implemented");
}

File: benchmark/problems/calculator/python/template.py
from __future__ import annotations
from dataclasses import dataclass
from enum import Enum, auto


class Op(Enum):
    ADD = auto()
    SUB = auto()
    MUL = auto()
    DIV = auto()


@dataclass
class Lit:
    value: float


@dataclass
class BinOp:
    op: Op
    left: "Expr"
    right: "Expr"


@dataclass
class Neg:
    inner: "Expr"


Expr = Lit | BinOp | Neg


def eval_expr(e: Expr) -> float:
    """Evaluate the expression. Raises ValueError on division by zero."""
    # TODO: implement
    raise NotImplementedError
